module-apps/apps-common/ApplicationCommon.cpp

```diff
@@ -948,6 +976,40 @@ namespace app
         return true;
     }
 
+    void ApplicationCommon::requestShutdownWindow(std::string windowName)
```

Comment:
and why exactly this and the following methods are needed?
Can't we just request for final refresh from the application scope via sending special "refresh" which is handled in: `handleAppShutdownRefresh` are these windows built somehow different than others? What do we gain from these methods?


Generally - why don't we handle only one, very specific refresh in the very specific window as a last.
i.e. if user declines to shutdown on request we still handle it as last refresh here?

Reply:
what do you mean by "if used declines to shutdown"? - user has nothing to say about that - the system is going down when battery is low and he cant do anything to stop that :)

> Why dont we just send special refresh to be handled in `handleAppShutdownRefresh`?
I do agree that we can do that. It was done like that in previous review, I guess I misunderstood the 3rd point of your first review note. Gonna rework that

Reply:
Please respond to all the questions :)